code/handlers/trackservers.q

```diff
@@ -333,13 +333,20 @@ reqprocsnotconn:{[requiredprocs]
     not all requiredprocs in exec proctype from .servers.SERVERS where .dotz.liveh[w]
   }
 
+startupdependent:{[requiredprocs;timeintv;cycles]
+  startupdepcycles[requiredprocs;timeintv;0W];   
+ }
+
 // Block process until all required processes are connected
-startupdependent:{[requiredprocs;timeintv]
-    while[.servers.reqprocsnotconn[requiredprocs];
-          .os.sleep[timeintv];
-          .servers.startup[]
-         ]
-  }
+startupdepcycles:{[requiredprocs;timeintv;cycles]
+  n:@[value;`n;0];                                                                                                      //variable used to check how many cycles have passed
+  while[.servers.reqprocsnotconn[requiredprocs];
+    n+:1;
+    if[n>cycles;.lg.e[`connectionreport;raze string[.proc.procname]," cannot connect to ",
+      $[1<count b:((),requiredprocs)except(),exec proctype from .servers.SERVERS where 
+      .dotz.liveh[w];","sv string@'b;string[b]]]]; 
```

Comment:
All the closing brackets make this a little confusing to read for me, I think it should read something like:
```
  while[.servers.reqprocsnotconn requiredprocs;
    n+:1
    if[n>cycles;
      b:((),requiredprocs)except(),exec proctype from .servers.SERVERS where .dotz.liveh w;
      .lg.e[`connectionreport;raze string[.proc.procname]," cannot connect to ",
        $[1<count b;","sv string@'b;string b]];
     ]; 
    .os.sleep timeintv;
    .servers.startup[];
   ];
```
I also think the if-else is redundant as `b` is always a list, if that's the case then `", "sv string b` should be fine in its place.